src/collector/Group.cpp

```diff
@@ -334,26 +338,14 @@ void Group::update_status(bool forbidden_dht)
             if (backend_ts < cur_ts)
                 backend_ts = cur_ts;
 
-            if (b_status == Backend::BAD) {
-                have_bad = true;
-                break;
-            } else if (b_status == Backend::RO) {
+            if (b_status == Backend::RO) {
```

Comment:
May be. Please file an issue if you think it's important. It is not related to this changeset.